### ISSUE TITLE
speed-transition.lua: Ignore subtitles with text shorter than a configurable amount.

### ISSUE THREAD

**Issue by CreamyCookie** [zenyd/mpv-scripts#20]
First of all, thanks for the `speed-transition.lua` script! It's really useful.

I don't know if this is possible, but it would be really useful if the script could ignore subtitles that are shorter than a certain length, since very short subtitles can be read quickly even at high speeds, and often they just contain something like `♪♫` when music is played.

Maybe it would be possible to use regex for that, so that one could also ignore subtitles where the whole text is enclosed in square brackets, which are often used to describe sounds and

**Comment by bitingsock** [comment]
I currently have a feature similar to this in a pending pull request. It is much simpler though. It is just a table of characters that if exist at all in the subtitle, it will will ignore that line.

**Comment by CreamyCookie** [comment]
@bitingsock Thanks a lot! This is great. 

In case it might be of use, I've converted my above regex into Lua's matching (I didn't know Lua had no regex, since I have basically no Lua experience) and added two characters from your change:

```lua

ignoredCharacters = "#♯♩♪♬♫🎵🎶"

-- Test strings:
s = "     [ loud noise ] " 
--s = " How are you doing?  "
--s = "  ♫ la deda ♪     "

local st = s:gsub("^%s*(.-)%s*$", "%1")
local ign = ignoredCharacters
local isLikelyMusicOrAudioSub = st:len() < 4 or st:find("^[" .. ign .. "]+.*[" .. ign .. "]+$") or st:find("^[ " .. ign .. "]+$") or st:find("^%[.*%]$")

print(isLikelyMusicOrAudioSub)
```

**Comment by bitingsock** [comment]
Cool, I'll try to implement this.

**Comment by bitingsock** [comment]
I got the pattern match working: https://github.com/bitingsock/mpv-scripts/blob/patch-4/speed-transition.lua
Waiting for pull approval, but you can use that one in the mean time.

**Comment by CreamyCookie** [comment]
@bitingsock Thanks a lot! =)

**Comment by CreamyCookie** [comment]
@bitingsock Just noticed something, and I think this was also the case previously.

When using subtitles that cannot easily be converted to text (e.g. image based ones), the script doesn't work at all. 

I'm not sure what the cause is, but sub-stepping seems to work.  You can step from one subtitle to the next. (Of course the match filter can't work without the actual text)

**Comment by bitingsock** [comment]
Correct, it has always only worked on Text bases subs. However, there may be some way to make a way using pure sub-stepping based properties that are already triggering the functions. I'll have to think about that.

**Comment by CreamyCookie** [comment]
That would be really awesome!

Not sure how useful, but one thing I've tried is to make `speed_transition` get triggered by a [`sub-start` property](https://mpv.io/manual/stable/#command-interface-sub-start) change (instead of `sub-text`, whose value is then manually requested). However, it seems that with image-based subtitles both `sub-start` and `sub-text` only trigger once, and then never again, which might be why it doesn't work. (Almost seems like a bug in mpv.. now that I think about it.)
